packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx

```diff
@@ -172,6 +178,13 @@ export const CollectionScreenDetailsTile = ({
   const firstTrack = useSelector(selectFirstTrack)
   const messages = getMessages(isAlbum ? 'album' : 'playlist')
   useRefetchLineupOnTrackAdd(collectionId)
+  const collection = useSelector((state) =>
+    typeof collectionId !== 'number'
+      ? 0
```

Comment:
I'm confused by this. Why is 0 the value we want here?

Reply:
tbh i am also confused by this. it's bc `collectionId` could be a `SmartCollectionVariant` which wouldn't work with the selector. i'm guessing in that case we wouldn't be using the purchase functionality so it's probably fine. cc @dylanjeffers .

Reply:
yeah my point was more that this selector as written is currently returning type `Collection | 0` instead of something like `Collection | null`

Reply:
sorry i didn't see this yall. turns out this is leading to a bug with search collections. will follow up with a fix.